src/components/UserInfo.jsx

```diff
@@ -126,15 +136,63 @@ function UserInfo({
     </section>
   );
 
+  const followUser = async () => {
+    try {
+      const response = await authenticatedFetch('follows/' + id, {
+        method: 'POST'
+      });
+
+      const newFollowing = [...user.following].concat([response]);
+      setUser({ ...user, following: newFollowing });
+      setIsFollowing(!isFollowing);
+      setFollowersCount(followersCount + 1);
+    } catch (e) {
+      console.log('ERROR:', e);
+    }
+  };
+
+  const unfollowUser = async () => {
+    try {
+      await authenticatedFetch('follows/' + id, { method: 'DELETE' });
+      const newFollowings = [...user.following].filter((u) => u.id !== id);
+      setUser({ ...user, following: newFollowings });
+      setIsFollowing(!isFollowing);
+      setFollowersCount(followersCount - 1);
+    } catch (e) {
+      console.log('ERROR: ', e);
+    }
+  };
```

Comment:
There's a bug--I spammed the follow button and it created an odd error. I'm now following the profile, but the button still says "Follow" and the follow count has gone up +1, and I can't unfollow.